projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts

```diff
@@ -112,15 +109,15 @@ export class CartesianLegend {
       .classed(`position-${legendPosition}`, true);
   }
 
-  private drawLegendEntry(element: EnterElement): Selection<HTMLDivElement, Series<{}>, null, undefined> {
-    const legendEntry = select<EnterElement, Series<{}>>(element).append('div').classed('legend-entry', true);
+  private drawLegendEntry(element: EnterElement): Selection<HTMLDivElement, Series<TData>, null, undefined> {
+    const legendEntry = select<EnterElement, Series<TData>>(element).append('div').classed('legend-entry', true);
 
     this.appendLegendSymbol(legendEntry);
     legendEntry
       .append('span')
       .classed('legend-text', true)
       .text(series => series.name)
-      .on('click', series => this.updateActiveSeries(series));
+      .on('click', series => (this.series.length > 1 ? this.updateActiveSeries(series) : undefined));
```

Comment:
do we need to change the css in this case too to remove the pointer?

Reply:
Yes, Updated this!!